lib/react_client/react_interop.dart

```diff
@@ -19,12 +19,12 @@ typedef ReactElement ReactJsComponentFactory(props, children);
 @JS()
 abstract class React {
   external static String get version;
+  external static ReactContext createContext(
+      [dynamic defaultValue, Function calculateChangedBits]);
```

Comment:
something like: 
`typedef int calculateChangedBits(dynamic currentValue, dynamic nextValue);`?

Reply:
Since Component2 will be Dart 2 only, we should be able to [use the new inline function types instead of typedefs!](https://www.dartlang.org/guides/language/effective-dart/design#prefer-inline-function-types-over-typedefs)

```dart
external static ReactContext createContext([
    dynamic defaultValue, 
    int Function(dynamic, dynamic) calculateChangedBits,
]);
```

Can `external`s be parameterized? If so, doing something like this could make the interface safer:

```dart
external static ReactContext createContext<T>([
    T defaultValue,
    int Function(T, T) calculateChangedBits,
]);
```

Reply:
I believe they can be, but I'm not sure if there's any runtime type-checking. +1 to generics.

Can we also add the parameter names?
```dart
int Function(T currentValue, T nextValue) calculateChangedBits,
```

Reply:
when i tried to parameterize it i got strange issues with the values passed to the actual js layer, i ended up with as the best i could do, I believe the dart version of the `createContext` function does have the parameterization though. 
```
external static ReactContext createContext([
    dynamic defaultValue,
    int Function(dynamic currentValue, dynamic nextValue) calculateChangedBits,
  ]);
```